packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -538,24 +545,36 @@ class NavigationIndicator extends StatelessWidget {
   /// If null, defaults to [ColorScheme.secondary].
   final Color? color;
 
-  /// The width of the container that holds in the indicator.
+  /// The width of this indicator.
   ///
   /// Defaults to `64`.
   final double width;
 
-  /// The height of the container that holds in the indicator.
+  /// The height of this indicator.
   ///
   /// Defaults to `32`.
   final double height;
 
-  /// The radius of the container that holds in the indicator.
+  /// The border radius of the shape of the indicator.
+  ///
+  /// This is used to create a [RoundedRectangleBorder] shape for the indicator.
+  /// This is ignored if [shape] is non-null.
   ///
   /// Defaults to `BorderRadius.circular(16)`.
   final BorderRadius borderRadius;
 
+  /// The shape of the indicator.
+  ///
+  /// If non-null this is used as the shape used to draw the background
+  /// of the indicator. If null then the value of [NavigationBarThemeData.shape]
+  /// is used. If that is null then a [RoundedRectangleBorder] with the
+  /// [borderRadius] is used.
+  final ShapeBorder? shape;
+
   @override
   Widget build(BuildContext context) {
-    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final NavigationBarThemeData theme = NavigationBarTheme.of(context);
+    final NavigationBarThemeData defaults = _defaultsFor(context);
```

Comment:
The `NavigationIndicator` is used for the `NavigationRail` too, so I think we want to avoid nav bar specific logic inside of here and rather pass it to the indicator widget instead

Reply:
Makes sense. Not sure what to do for defaults here if these parameters aren't given. These are optional parameter to the Indicator, so if someone is using it (outside of our nav rail or bar), we would still need defaults. We could make them required fields, but that would be a possibly breaking change.

Reply:
Ok, I have pulled out the default logic from the Indicator. However, this may still not be the best as it just got moved to the NavigationDestination as that is what creates the indicators. While that isn't currently shared, it will hopefully be used by other navigation widgets in the future.